lib/typesafe.h

```diff
@@ -309,7 +309,8 @@ static inline void typesafe_dlist_add(struct dlist_head *head,
 		struct dlist_item *prev, struct dlist_item *item)
 {
 	item->next = prev->next;
-	item->next->prev = item;
+	if (item->next)
```

Comment:
What's the SA warning here? `DLIST` is a full ring with the listhead, it should be impossible to run into `NULL` here. I guess the SA doesn't understand that?

Reply:
(regardless, this shouldn't be mixed in with a completely unrelated PR)

Reply:
hmm: are you saying ... "I wish this weren't here in this PR", or "get this out of this PR" ?

Reply:
interesting - so clang 14 (in an ubuntu 22 vm) didn't complain, but clang-11 in the CI does.

Reply:
> interesting - so clang 14 (in an ubuntu 22 vm) didn't complain, but clang-11 in the CI does.

Ah I see it now… taking a look.

Reply:
It seems to think that since `.next` is never written (bunch of `Returning without writing to 'item->link.di.next'`), it might be `NULL`… but it can't be.

Not sure what to do here, if anything an assertion might be useful, or maybe a `if (next == NULL) __builtin_unreachable();`

Reply:
so extremely fun - can't just use 'assert', because the xref lib code redefines it - and that in turn uses typesafe.
I think the test is pretty easy to understand, even if it's always 'true' - I'll open a PR with that one-liner.